catalog/services_state.go

```diff
@@ -514,21 +512,7 @@ func (state *ServicesState) EachService(fn func(hostname *string, serviceId *str
 // Return a properly regex-matched name for the service, or failing that,
 // the Image ID which we use to stand in for the name of the service.
 func (state *ServicesState) ServiceName(svc *service.Service) string {
-	var svcName string
-
-	if state.ServiceNameMatch != nil {
-		toMatch := []byte(svc.Name)
-		matches := state.ServiceNameMatch.FindSubmatch(toMatch)
-		if len(matches) < 1 {
-			svcName = svc.Image
-		} else {
-			svcName = string(matches[1])
-		}
-	} else {
-		svcName = svc.Image
-	}
-
-	return svcName
+	return svc.Name
```

Comment:
I would suggest we just eliminate this function entirely in this case. Find/replace with `svc.Name`.